Feat/more attribute support

secDre4mer opened the pull request:
<!---
Please provide answers in the spaces below each prompt, where applicable.
Not every PR requires responses for each prompt.
Use your discretion.
-->
#### Name of feature: 

Support for reading / changing unsigned attributes

#### Pain or issue this feature alleviates:

Unsigned attributes can contain mutable information. E.g. adding a 2nd signature to a PKCS7 struct is done by adding an additional unsigned attribute that contains the PKCS7 for the secondary signature.

#### Why is this important to the project (if not answered above):

Reading and changing unsigned signatures is a valuable extension, especially since support for adding unsigned signatures at signing time already exists.

#### Is there documentation on how to use this feature? If so, where?

#### In what environments or workflows is this feature supported?

#### In what environments or workflows is this feature explicitly NOT supported (if any)?

#### Supporting links/other PRs/issues:

https://github.com/saferwall/pe/pull/95

💔Thank you!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/smallstep/pkcs7?pullRequest=22) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/smallstep/pkcs7?pullRequest=22) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/smallstep/pkcs7?pullRequest=22) it.</sub>

hslatman commented:
Hey @secDre4mer, thank you for opening the PR. I'll have a look at this soon.

I'm curious about the `Copy` function name / signature. It seems like the goal is to get an exact copy of an existing `p7` struct, but then with specific unsigned attributes set? Maybe it would be more logical to define it as a method on the `p7` itself, and return a new instance? Or with a more explicit name, like `CopyWithUnsignedAttributes`?

secDre4mer commented:
Hey @hslatman , yes, you've described the use case well (basically: I want to add an unwanted attribute

secDre4mer commented:
Added `CopyWithUnsignedAttributes` as suggested by you. Also added a `Marshal` function that can be used to turn a `PKCS7` struct back into an ASN1 byte struct.

LordNoteworthy commented:
Thank @secDre4mer for the update ! 

What do you think @hslatman  ?

secDre4mer commented:
Just to bump this: @hslatman is this still under consideration?

hslatman commented:
Hey @secDre4mer, yes, it is. I need some time to get to this properly.
